pkg/api/addons.go

```diff
@@ -352,6 +352,12 @@ func assignDefaultAddonVals(addon, defaults KubernetesAddon, isUpdate bool) Kube
 	if addon.Enabled == nil {
 		addon.Enabled = defaults.Enabled
 	}
+	if !to.Bool(addon.Enabled) {
+		return KubernetesAddon{
```

Comment:
Should we add something like this:

```
if addon.Containers != nil || addon.Config != nil || addon.Data != nil {
    log.Warn("configuration data for disabled addon <addon> will be ignored and removed from generated api model")
}
```

Something like that?

Reply:
hmm the thing is that at this point the config/containers/etc won't be nil if the addon wasn't specified by the user in their apimodel. ie. If I omit autoscaler from my apimodel, the `appendAddonIfNotPresent` function will append the whole addon default struct. This means that the warning will be logged for every single addon that was omitted. The only way that warning message would be accurate would be if I specified autoscaler in my apimodel but set it to disabled. That or we need to refactor the way default addons are set when they are omitted from the apimodel which IMO is more effort than this small UX improvement was meant to take.

Reply:
Got it